EKF/ekf_helper.cpp

```diff
@@ -1491,6 +1488,49 @@ void Ekf::updateBaroHgtOffset()
 	}
 }
 
+Vector3f Ekf::getVisionVelocityInEkfFrame()
+{
+	// correct velocity for offset relative to IMU
+	const Vector3f pos_offset_body = _params.ev_pos_body - _params.imu_pos_body;
+	const Vector3f vel_offset_body = _ang_rate_delayed_raw % pos_offset_body;
+
+	// rotate measurement into correct earth frame if required
+	switch(_ev_sample_delayed.vel_frame) {
+		case BODY_FRAME_FRD:
+			return _R_to_earth * (_ev_sample_delayed.vel - vel_offset_body);
+
+		case LOCAL_FRAME_FRD:
+			const Vector3f vel_offset_earth = _R_to_earth * vel_offset_body;
+			if(_params.fusion_mode & MASK_ROTATE_EV)
+			{
+				return _R_ev_to_ekf *_ev_sample_delayed.vel - vel_offset_earth;
+			}
+			return _ev_sample_delayed.vel - vel_offset_earth;
+	}
+	// switch statement guarantees to return
```

Comment:
Does it though? What if an invalid vel_frame is specified?

Reply:
Yes. I checked. As long as there is no default case, the compiler will throw an error if there is a label missing or a label that does not belong to the enum.